healthcheck/healthcheck.go

```diff
@@ -0,0 +1,216 @@
+// This package provides a simple health check server.
+//
+// The package lets you register arbitrary health check endpoints by
+// providing a name (== URL path) and a callback function.
+//
+// The health check server listens for HTTP requests on a given port;
+// probes are executed by issuing requests to the endpoints' named
+// URL paths.
+//
+// GETing the "/" path provides a list of registered endpoints, one
+// per line. GETing "/_ALL_" probes each registered endpoint sequentially,
+// returning each endpoint's path, HTTP status code and body per line.
+//
+// The package works as a "singleton" with just one server in order to
+// avoid cluttering the main program by passing handles around.
+package healthcheck
```

Comment:
A package just for one file seems a bit overblown for me.
Sure, core is overloaded at the moment but the package should at least be core/healthcheck.
If it is totally generic it should go to github.com/trivago/tgo